NASA-PDS/validate
# Handle leading zeros values, accept a single-character filename and correct file separator for non-linux machine

Closes NASA-PDS/validate#297, NASA-PDS/validate#299 and NASA-PDS/validate#300

This pull request resolves three issues:

1. Values in tabulated file with leading zeros are read correctly for real and integer numbers.
2. Accepts a file name with a single character.
3. Corrects the file separator for non-linus machine when the registered context product is updated.

Tested in DEV:

To test NASA-PDS/validate#297


% validate --skip-context-validation -R pds4.label -r report_github297_label_valid.json   -s json -t src/test/resources/github297/valid/rimfax_rdr_0081_example.xml
% validate --skip-context-validation -R pds4.label -r report_github297_label_invalid.json   -s json -t src/test/resources/github297/invalid/rimfax_rdr_0081_example.xml

To test NASA-PDS/validate#299

% validate -R pds4.label -r report_github299_label_valid.json   -s json -t src/test/resources/github299/valid/gbo_ast_fieber-beyer_spectra_v2.0_20210211_aip_v1.0.xml
% validate -R pds4.label -r report_github299_label_invalid.json -s json -t src/test/resources/github299/invalid/gbo_ast_fieber-beyer_spectra_v2.0_20210211_aip_v1.0.xml

To test NASA-PDS/validate#300:

First make a copy of your registered_context_products.json file

% cp src/main/resources/util/registered_context_products.json src/main/resources/util/registered_context_products.json.original.03_09_2021
% validate -u

The above command should replace your current registered_context_products.json with the latest from the server.
Reminder that running regression test at this point may fail.
